src/jmh/java/com/palantir/ri/ResourceIdentifierBenchmark.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.ri;
+
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.profile.GCProfiler;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.TimeValue;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 3, time = 3, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 3, time = 3, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@Threads(4)
+@SuppressWarnings({"checkstyle:hideutilityclassconstructor", "VisibilityModifier", "DesignForExtension"})
+public class ResourceIdentifierBenchmark {
+
+    public enum Input {
+        NO_INSTANCE("ri.my-service..environment.8c51aa27-32f6-4d25-a3a5-966196e57be3"),
+        UUID("ri.apollo.main.installation.8c51aa27-32f6-4d25-a3a5-966196e57be3");
+
+        private final String string;
+
+        Input(String string) {
+            this.string = string;
+        }
+    }
+
+    @Param
+    Input input;
```

Comment:
I _think_ this can be simplified to the following, but it's possible I'm misremembering:

```suggestion
    @Param({
        "ri.my-service..environment.8c51aa27-32f6-4d25-a3a5-966196e57be3", 
        "ri.apollo.main.installation.8c51aa27-32f6-4d25-a3a5-966196e57be3"})
    String input;
```

Reply:
Yep this does work perfectly - I just kinda liked the enum for readability of the result tables:

```
ResourceIdentifierBenchmark.fromString:·gc.alloc.rate                    ri.apollo.main.installation.8c51aa27-32f6-4d25-a3a5-966196e57be3  avgt    3  1078.746 ± 338.169  MB/sec
```
vs
```
ResourceIdentifierBenchmark.fromString:·gc.alloc.rate                           UUID  avgt    3   982.036 ±  395.425  MB/sec
```